server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityHealthIndicatorServiceTests.java

```diff
@@ -1532,6 +1534,150 @@ public void testLimitNumberOfAffectedResources() {
         }
     }
 
+    public void testShouldBeGreenWhenFrozenIndexIsUnassignedAndOriginalIsAvailable() {
+        String originalIndex = "logs-2023.07.11-000024";
+        String restoredIndex = "restored-logs-2023.07.11-000024";
+        var clusterState = createClusterStateWith(
+            List.of(
+                IndexMetadata.builder(restoredIndex)
+                    .settings(
+                        Settings.builder()
+                            .put(IndexMetadata.SETTING_VERSION_CREATED, IndexVersion.current())
+                            .put(SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_INDEX_NAME_SETTING_KEY, originalIndex)
+                            .put(IndexModule.INDEX_STORE_TYPE_SETTING.getKey(), SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                            .put(SearchableSnapshotsSettings.SEARCHABLE_SNAPSHOT_PARTIAL_SETTING_KEY, randomBoolean())
+                            .build()
+                    )
+                    .numberOfShards(1)
+                    .numberOfReplicas(0)
+                    .build(),
+                IndexMetadata.builder(originalIndex)
+                    .settings(Settings.builder().put(IndexMetadata.SETTING_VERSION_CREATED, IndexVersion.current()).build())
+                    .numberOfShards(1)
+                    .numberOfReplicas(0)
+                    .build()
+            ),
+            List.of(
+                index(restoredIndex, new ShardAllocation(randomNodeId(), UNAVAILABLE)),
+                index(originalIndex, new ShardAllocation(randomNodeId(), AVAILABLE))
+            ),
+            List.of(),
+            List.of()
+        );
+        var service = createShardsAvailabilityIndicatorService(clusterState);
+
+        HealthIndicatorResult result = service.calculate(true, HealthInfo.EMPTY_HEALTH_INFO);
+        assertThat(
+            result,
+            equalTo(
+                createExpectedResult(
+                    GREEN,
+                    "This cluster has 1 unavailable primary shard.",
+                    Map.of("unassigned_primaries", 1, "started_primaries", 1),
+                    List.of(),
+                    List.of(
+                        new Diagnosis(ACTION_CHECK_ALLOCATION_EXPLAIN_API, List.of(new Diagnosis.Resource(INDEX, List.of(restoredIndex))))
+                    )
```

Comment:
Is this at risk of causing confusion? I don't think GREEN should report problems, but perhaps the symptom should describe that even though we have one unavailable primary, it's a mounted index and the original index is still present?

Reply:
I added one more sentence in case the only unavailable shards are the ones from the "green' frozen index. I think we can improve this in follow up work since we will probably fine grain the impacts as a bit more.